### PR TITLE
docs: add release notes for 0.30.0

### PR DESCRIPTION
**Pull request by wes-turner**
Add release notes for 0.30.0.

### CODE REVIEW

File: docs/release-notes.rst [hunk 1]
```diff
@@ -6,6 +6,53 @@
...
+**Release Date:** April 04, 2024
+
+**Breaking Changes**
+
+-  API: :class:`~determined.pytorch.Trainer` no longer supports the ``Trainer.configure_profiler``
+   option. Profiling is now enabled through the ``Trainer.fit(profiling_enabled=True)`` call.
+
+-  Database migration: system metrics collected by the Determined profiler are now stored in the
+   generic ``metrics`` table. This requires a few schema changes to the ``metrics`` table that will
+   be run during migrations.
```
Comment (tara-det-ai) [inline docs/release-notes.rst:25]: -  Database migration: System metrics collected by the Determined profiler are now stored in the
   generic ``metrics`` table. This requires a few schema changes to the ``metrics`` table that will
   be run during migrations.

File: docs/release-notes.rst [hunk 2]
```diff
@@ -6,6 +6,53 @@
...
+
+.. important::
+
+   This migration will be more time-consuming for deployments with a large amount of metrics stored.
+
```
Comment (tara-det-ai) [inline docs/release-notes.rst:30]: .. important::

   This migration may take more time for deployments with a large amount of stored metrics.

File: docs/release-notes.rst [hunk 3]
```diff
@@ -6,6 +6,53 @@
...
+**New Features**
+
+-  Core API: the Determined profiler is now accessible from the Core API. It collects system
+   metrics, which can be viewed in the Web UI under the experiment's "Profiler" tab. See the
```
Comment (tara-det-ai) [inline docs/release-notes.rst:34]: WebUI (rather than "Web UI")

File: docs/release-notes.rst [hunk 4]
```diff
@@ -6,6 +6,53 @@
...
+   :ref:`Core API guide <core-profiler>` for details.
```
Comment (tara-det-ai) [inline docs/release-notes.rst:35]: Does this page exist?

File: docs/release-notes.rst [hunk 5]
```diff
@@ -6,6 +6,53 @@
...
+
+**Removed Features**
+
+-  Profiler: support for timing metrics and related configurations have been removed. The Determined
+   profiler now only collects system metrics and defers to our native profiler integrations for
+   training-specific profiling. Users are encouraged to configure profilers native to their
+   :ref:`training API <apis-howto-overview>` for this functionality.
+
+   -  Historical data for timing metrics is retained in the ``trial_profiler_metrics`` database
+      table, but they are no longer being collected or rendered in the Web UI.
```
Comment (tara-det-ai) [inline docs/release-notes.rst:45]: WebUI

File: docs/release-notes.rst [hunk 6]
```diff
@@ -6,6 +6,53 @@
...
+
+   -  Historical data for system metrics generated by trials before this release are not
+      automatically migrated due to time cost. For users wanting to view historical system metrics
+      in the Web UI, we provide an `optional migration script
```
Comment (tara-det-ai) [inline docs/release-notes.rst:49]: WebUI

**Review by tara-det-ai — APPROVED** [review]
LGTM, added suggestions for consistency

File: docs/release-notes.rst [hunk 7]
```diff
@@ -6,6 +6,53 @@
...
+**Breaking Changes**
+
+-  API: :class:`~determined.pytorch.Trainer` no longer supports the ``Trainer.configure_profiler``
+   option. Profiling is now enabled through the ``Trainer.fit(profiling_enabled=True)`` call.
+
+-  Database migration: system metrics collected by the Determined profiler are now stored in the
+   generic ``metrics`` table. This requires a few schema changes to the ``metrics`` table that will
+   be run during migrations.
+
+.. important::
```
Comment (dzhu) [inline docs/release-notes.rst:27]: Indent so it's part of the list item.

File: docs/release-notes.rst [hunk 8]
```diff
@@ -6,6 +6,53 @@
...
+
+**New Features**
+
+-  Core API: the Determined profiler is now accessible from the Core API. It collects system
+   metrics, which can be viewed in the Web UI under the experiment's "Profiler" tab. See the
+   :ref:`Core API guide <core-profiler>` for details.
+
+**Removed Features**
+
+-  Profiler: support for timing metrics and related configurations have been removed. The Determined
```
Comment (dzhu) [inline docs/release-notes.rst:39]:
```suggestion
-  Profiler: Support for timing metrics and related configurations has been removed. The Determined
```

File: docs/release-notes.rst (hunk 1 above)
Comment (dzhu) [reply docs/release-notes.rst:25]: Others too, don't forget.

**Review by dzhu — APPROVED** [review]

File: docs/release-notes.rst (hunk 1 above)
Comment (wes-turner) [reply docs/release-notes.rst:25]: Capitalized!

File: docs/release-notes.rst (hunk 2 above)
Comment (wes-turner) [reply docs/release-notes.rst:30]: Thanks!

File: docs/release-notes.rst (hunk 8 above)
Comment (wes-turner) [reply docs/release-notes.rst:39]: nice catch!

File: docs/release-notes.rst (hunk 7 above)
Comment (wes-turner) [reply docs/release-notes.rst:27]: done

File: docs/release-notes.rst (hunk 4 above)
Comment (wes-turner) [reply docs/release-notes.rst:35]: Yes! It points here: model-dev-guide/api-guides/apis-howto/api-core-ug.html#core-profiler

**Pull request merged**